main.tf

```
@@ -1,6 +1,8 @@
 locals {
   principals_readonly_access_non_empty = length(var.principals_readonly_access) > 0 ? true : false
+  principals_readonly_access_lambda    = length(var.principals_lambda) > 0 ? true : false
   principals_full_access_non_empty     = length(var.principals_full_access) > 0 ? true : false
+  principals_full_access_lambda        = length(var.principals_lambda) > 0 ? true : false
```

you're probably bring consistent but this can be simplified by omitting the bools since the conditional is truthy
```suggestion
  principals_full_access_lambda        = length(var.principals_lambda) > 0
```